include/execution.hpp

```diff
@@ -313,49 +318,76 @@ namespace std::execution {
-            __allocator<tag_invoke_result_t<get_allocator_t, __cref_t<_Receiver>>>
-        tag_invoke_result_t<get_allocator_t, __cref_t<_Receiver>> operator()(_Receiver&& __rcvr) const
-          noexcept(nothrow_tag_invocable<get_allocator_t, __cref_t<_Receiver>>) {
-          return tag_invoke(get_allocator_t{}, std::as_const(__rcvr));
+        template <class _T>
+          requires nothrow_tag_invocable<get_allocator_t, __cref_t<_T>> &&
+            __allocator<tag_invoke_result_t<get_allocator_t, __cref_t<_T>>>
+        tag_invoke_result_t<get_allocator_t, __cref_t<_T>> operator()(_T&& __t) const
+          noexcept(nothrow_tag_invocable<get_allocator_t, __cref_t<_T>>) {
+          return tag_invoke(get_allocator_t{}, std::as_const(__t));
         }
       };
 
       struct get_stop_token_t {
-        template <receiver _Receiver>
-          requires tag_invocable<get_stop_token_t, __cref_t<_Receiver>> &&
-            stoppable_token<tag_invoke_result_t<get_stop_token_t, __cref_t<_Receiver>>>
-        tag_invoke_result_t<get_stop_token_t, __cref_t<_Receiver>> operator()(_Receiver&& __rcvr) const
-          noexcept(nothrow_tag_invocable<get_stop_token_t, __cref_t<_Receiver>>) {
-          return tag_invoke(get_stop_token_t{}, std::as_const(__rcvr));
+        template <class _T>
+          requires tag_invocable<get_stop_token_t, __cref_t<_T>> &&
+            stoppable_token<tag_invoke_result_t<get_stop_token_t, __cref_t<_T>>>
+        tag_invoke_result_t<get_stop_token_t, __cref_t<_T>> operator()(_T&& __t) const
+          noexcept(nothrow_tag_invocable<get_stop_token_t, __cref_t<_T>>) {
+          return tag_invoke(get_stop_token_t{}, std::as_const(__t));
         }
-        never_stop_token operator()(receiver auto&&) const noexcept {
+        never_stop_token operator()(auto&&) const noexcept {
           return {};
         }
       };
-    }
+    } // namespace __impl
+
     using __impl::get_allocator_t;
     using __impl::get_scheduler_t;
     using __impl::get_stop_token_t;
     inline constexpr get_scheduler_t get_scheduler{};
     inline constexpr get_allocator_t get_allocator{};
     inline constexpr get_stop_token_t get_stop_token{};
-  }
+  } // namespace __general_queries
 
-  template <class _Receiver>
+  template <class _T>
     using stop_token_of_t =
-      remove_cvref_t<decltype(get_stop_token(__declval<_Receiver>()))>;
+      remove_cvref_t<decltype(get_stop_token(__declval<_T>()))>;
+
+  // [execution.receivers.queries], receiver queries
+  inline namespace __receiver_queries {
+    namespace __impl {
+      struct forwarding_receiver_query_t {
+        template <class _Tag>
+        constexpr bool operator()(_Tag __tag) const noexcept {
+          if constexpr (tag_invocable<forwarding_receiver_query_t, _Tag>) {
+            return tag_invoke(*this, (_Tag&&) __tag);
+          } else {
+            return __none_of<_Tag, set_value_t, set_error_t, set_done_t>;
```

Comment:
I expected this to return false if there is no customization- this looks like it returns true. Am I confused?

Edit: Ah I see that it is replacing the previous usage of the none_of expression. This would mean that every cpo defaults to being forwarded on a receiver and the only customizations are to prevent being forwarded on a receiver.

I am conflicted but don't have a strong reason to flip the excludelist to a includelist. I have a discomfort with excludelists from my time in Windows security. They are generally considered a mistake in the security space.